api.php

```diff
@@ -0,0 +1,211 @@
+<?php
+ini_set('display_errors', 1);
+
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Slim\Factory\AppFactory;
+
+
+require __DIR__ . '/vendor/autoload.php';
+require __DIR__ . '/src/User.php';
+require __DIR__ . '/src/Review.php';
+
+// var_dump($_SERVER);
+/**
+ * Instantiate App
+ *
+ * In order for the factory to work you need to ensure you have installed
+ * a supported PSR-7 implementation of your choice e.g.: Slim PSR-7 and a supported
+ * ServerRequest creator (included with Slim PSR-7)
+ */
+$app = AppFactory::create();
+
+// Add Routing Middleware
+$app->addRoutingMiddleware();
+
+/**
+ * Add Error Handling Middleware
+ *
+ * @param bool $displayErrorDetails -> Should be set to false in production
+ * @param bool $logErrors -> Parameter is passed to the default ErrorHandler
+ * @param bool $logErrorDetails -> Display error details in error log
+ * which can be replaced by a callable of your choice.
+
+ * Note: This middleware should be added last. It will not handle any exceptions/errors
+ * for middleware added after it.
+ */
+$errorMiddleware = $app->addErrorMiddleware(true, true, true);
+
+// Define app routes
+$app->get('/test', function (Request $request, Response $response, $args) {
+	$response->getBody()->write(json_encode($_SERVER));
+	return $response;
+});
+
+/********* USER ROUTES *********/
+
+// lists a users info including their reviews //
+$app->get('/user/{id}', function (Request $request, Response $response, array $args) {
+	$id = $args['id'];
+	$user = User::findUserById($id);
+	if(empty($user)){
+		$response->getBody()->write('user not found');
+		return $response->withStatus(404);
+	}
+	//returning users info
+	$response->getBody()->write(json_encode($user->jsonSerialize()));
+	return $response;
+});
+
+// Create new user
+$app->post('/users/new', function (Request $request, Response $response, array $args) {
+	$body = json_decode($request->getBody());
+
+	$user = new User($body->first_name, $body->last_name, $body->email, $body->password, $body->phone, $body->preferred ?? false);
+	$response->getBody()->write(json_encode($user));
```

Comment:
this needs the be saved to the database, as it stands it does not save to db

Reply:
i thought _you_ fixed that!?

Reply:
did you not push up the changes we made? I **know** I watched you write out the `saveToDB()` function cuz you had to look up the correct spelling.

Reply:
we fixed on review not user!

Reply:
on it

Reply:
done